internal/scout/resources/resources_test.go

```diff
@@ -0,0 +1,168 @@
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "Error calling ResourcesK8s"))
+	}
+}
+
+type DockerClientMock struct {
+	mock.Mock
+}
+
+func (m *DockerClientMock) ContainerList(ctx context.Context, options types.ContainerListOptions) ([]types.Container, error) {
+	args := m.Called(ctx, options)
+	return args.Get(0).([]types.Container), args.Error(1)
+}
+
+func (m *DockerClientMock) ContainerInspect(ctx context.Context, containerID string) (types.ContainerJSON, error) {
+	args := m.Called(ctx, containerID)
+	return args.Get(0).(types.ContainerJSON), args.Error(1)
+}
+
+func (m *DockerClientMock) Close() error {
+	args := m.Called()
+	return args.Error(0)
+}
+
+func TestResourcesDocker(t *testing.T) {
+	dockerClient := new(DockerClientMock)
+
+	dockerClient.On("ContainerList", mock.Anything, mock.Anything).Return([]types.Container{
+		{ID: "container1"},
+		{ID: "container2"},
+		{ID: "container3"},
+	}, nil)
+
+	dockerClient.On("ContainerInspect", mock.Anything, "container1").Return(types.ContainerJSON{
+		ContainerJSONBase: &types.ContainerJSONBase{
+			Name: "container1",
+			HostConfig: &container.HostConfig{
+				Resources: container.Resources{
+					NanoCPUs:          2000000000,
+					CPUShares:         512,
+					Memory:            1536870912,
+					MemoryReservation: 268435456,
+				},
+			},
+		},
+	}, nil)
+
+	dockerClient.On("ContainerInspect", mock.Anything, "container2").Return(types.ContainerJSON{
+		ContainerJSONBase: &types.ContainerJSONBase{
+			Name: "container2",
+			HostConfig: &container.HostConfig{
+				Resources: container.Resources{
+					NanoCPUs:          1000000000,
+					CPUShares:         1024,
+					Memory:            268435456,
+					MemoryReservation: 134217728,
+				},
+			},
+		},
+	}, nil)
+
+	dockerClient.On("ContainerInspect", mock.Anything, "container3").Return(types.ContainerJSON{
+		ContainerJSONBase: &types.ContainerJSONBase{
+			Name: "container3",
+			HostConfig: &container.HostConfig{
+				Resources: container.Resources{
+					NanoCPUs:          4000000000,
+					CPUShares:         2048,
+					Memory:            5268435456,
+					MemoryReservation: 4134217728,
+				},
+			},
+		},
+	}, nil)
+
+	dockerClient.On("Close").Return(nil)
+
+	var expectedOutput strings.Builder
+	expectedW := tabwriter.NewWriter(&expectedOutput, 0, 0, 2, ' ', 0)
+
+	fmt.Fprintln(expectedW, "Container\tCPU Cores\tCPU Shares\tMem Limits\tMem Reservations")
+	fmt.Fprintf(expectedW, "container1\t2\t512\t1 GB\t268 MB\n")
+	fmt.Fprintf(expectedW, "container2\t1\t1024\t268 MB\t134 MB\n")
+	fmt.Fprintf(expectedW, "container3\t4\t2048\t5 GB\t4 GB\n")
+	expectedW.Flush()
+
+	oldStdout := os.Stdout
+	r, w, _ := os.Pipe()
+	os.Stdout = w
+
+	err := Docker(context.Background(), dockerClient)
+	if err != nil {
+		t.Fatalf("ResourcesDocker returned an error: %v", err)
+	}
+
+	err = dockerClient.Close()
+	if err != nil {
+		t.Fatalf("Error closing docker client: %v", err)
+	}
+
+	w.Close()
```

Comment:
Even though this is just a test, we probably still want to use a `defer` with this

Reply:
Hmm... the test totally freaked out when I deferred this for some reason.

Reply:
Keeping it the same for now.